src/Google/Ads/GoogleAds/Lib/V1/ServiceClientFactoryTrait.php

```diff
@@ -164,6 +164,17 @@ public function getGoogleAdsClientOptions()
             ];
         }
 
+        $clientOptions['transportConfig']['grpc'] += [
```

Comment:
What will happen here if `$clientOptions['transportConfig']['grpc']` is not set when this is called? (Which presumably could happen if `$this->getLogger()` returned empty above?)

Unfortunately there isn't a nice, clean way of merging PHP arrays (that I know of) that will avoid wiping out possible existing configuration. One approach might be to set this setting *before* the conditional setting of intercepts, because this will always be set. So your code might look like:

```php
$clientOptions['transportConfig'] = [
  'grpc' => [
    'stubOpts' => [
      // ...
    ]
  ]
];
if (!empty($this->getLogger())) {
  // ...
  $clientOptions['transportConfig']['grpc'] += ['interceptors' => [$googleAdsLoggingInterceptor]];
}
```

Reply:
Thanks for spotting this.